src/components/button/button.tsx

```diff
@@ -102,7 +102,7 @@ export interface EuiButtonProps extends CommonProps {
 }
 
 export interface EuiButtonDisplayProps extends EuiButtonProps {
-  element: 'a' | 'button' | 'label';
+  element: keyof JSX.IntrinsicElements;
```

Comment:
Out of curiosity, what is this/does this mean?

Reply:
Also, would `React.createElement` help here in any way?

Reply:
> keyof JSX.IntrinsicElements

It will allow for basically any HTML element

> Also, would React.createElement help here in any way?

That's an interesting idea. Let me try it out; I'm not actually sure what TS will do it.